### ISSUE TITLE
Add ability to define dependency groups

### ISSUE THREAD

**Issue by valentincalomme** [seddonym/import-linter#231]
My use case is the following. My package has separate submodules (i.e. `cli`, `app`, `etl`, `core`) and I want to keep a strict separation of concerns.

I want the `app` module to be the only module that can import "app" dependencies (i.e. `fastapi`), my `cli` module should be the only one capable of importing "cli" dependencies (i.e. `click`).

This is currently achievable by simply creating a forbidden contract type. In some cases, the group of dependencies I'd like to exclude is relatively large, and I would like to reuse that group of dependencies in multiple places (i.e. in my `app` module and `tests/app` module).

It would be great to be able to define "dependency groups" in the `import-linter` configuration. Something like that:

```toml
[importlinter.groups]

cli = ["typer"]
app = ["fastapi"]
docs = ["mkdocs", "colorama"]
```
So that contracts could look like this:

```toml

[importlinter:contracts:test-dependencies-forbidden-in-other-modules]
name = Test dependencies may not be imported by other modules than the tests module.
type = forbidden
source_modules =
    my_package
forbidden_modules =
    app
ignore_imports =
    my_package.app.* -> app
```

There is a currently PEP in draft (https://peps.python.org/pep-0735/) that would allow these groups to be defined in a `pyproject.toml`, and it would be even better if you

**Comment by seddonym** [comment]
Thanks for the issue!

Interesting idea. My concern is that it adds complexity to Import Linter, so I'm leaning towards saying 'probably not' for the moment. However if there were a lot more requests for this feature from different people then I might warm up to it 😄 . So, leaving this open for now.

**Comment by valentincalomme** [comment]
Yes, it's definitely much more of a nice to have! Just wanted to document it
